src/responder/nss/nsssrv_cmd.c

```diff
@@ -4573,8 +4623,19 @@ static errno_t nss_cmd_getsidby_search(struct nss_dom_ctx *dctx)
                 req_type = SSS_DP_USER_AND_GROUP;
             }
 
+            if (cmdctx->name_is_upn) {
+                extra_flag = EXTRA_NAME_IS_UPN;
+            } else if (DOM_HAS_VIEWS(dom) && (dctx->res->count == 0
+                    || ldb_msg_find_attr_as_string(dctx->res->msgs[0],
+                                                   OVERRIDE_PREFIX SYSDB_NAME,
+                                                   NULL) != NULL)) {
```

Comment:
I started writing tests for this new functionality and I wonder if this is another a bit unrelated thing this patch fixes -- looking up SID by overriden name?